File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums=set(nums)
        maxLen = 0 

        for num in nums:
            curnum = num
            curlen = 1 
            if curnum-1 not in nums:
                while curnum+1 in nums:
                    curnum = curnum+1
                    curlen+=1
                maxLen = max(maxLen, curlen)
        return maxLen
        

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> set;
        for(int i=0;i<nums.size();i++)
            set.insert(nums[i]);
        
        int curLen = 0 ;
        int maxLen = 0;
        int curNum;
        for(auto it : set)
        {
            curNum = it;
            if(!set.count(curNum-1)){
                curLen = 1;
                while(set.count(curNum+1)){
                    curLen++;
                    curNum++;
                    
                }
                maxLen = max(maxLen, curLen);
            }
        }
        return maxLen;

        
    }
};